tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -2543,6 +2546,69 @@ uint32_t SPIRVEmitter::processIntrinsicCallExpr(const CallExpr *callExpr) {
   return processIntrinsicUsingGLSLInst(callExpr, glslOpcode, actOnEachVecInMat);
 }
 
+uint32_t SPIRVEmitter::processIntrinsicClamp(const CallExpr *callExpr) {
+  // According the HLSL reference: clamp(X, Min, Max) takes 3 arguments. Each
+  // one may be int, uint, or float.
+  // If *any* of the 3 arguments is a float, clamp will have the following
+  // signature: 'float (float, float, float)'. ImplicitCastExpr is automatically
+  // added to handle conversions.
+  // If there are no float arguments, if *any* argument is an unsigned integer,
+  // the following signature is used:
+  // 'unsigned int (unsigned int, unsigned int, unsigned int)'
+  // ImplicitCastExpr is automatically added to handle conversions.
+  // If all 3 arguments are signed integers, then 'int (int, int, int)'
+  // signature is used.
+
+  const uint32_t glslInstSetId = theBuilder.getGLSLExtInstSet();
+  GLSLstd450 glslOpcode = GLSLstd450Bad;
+  const QualType returnType = callExpr->getType();
+  const uint32_t returnTypeId =
+      typeTranslator.translateType(callExpr->getType());
+
+  // Get the function parameters. Expect 3 parameters.
+  assert(callExpr->getNumArgs() == 3u);
+  const Expr *argX = callExpr->getArg(0);
+  const Expr *argMin = callExpr->getArg(1);
+  const Expr *argMax = callExpr->getArg(2);
+  const QualType argXType = argX->getType();
+  const QualType argMinType = argMin->getType();
+  const QualType argMaxType = argMax->getType();
+  if (isFloatOrVecMatOfFloatType(argXType) ||
```

Comment:
I think we only need to check the return type of the function call? The front end should've selected the correct overload already.